account_financial_risk/models/res_partner.py

```diff
@@ -219,14 +202,15 @@ def _risk_account_groups(self):
     def _compute_risk_account_amount(self):
         self.update(
             {
+                "risk_invoice_draft": 0.0,
                 "risk_invoice_open": 0.0,
                 "risk_invoice_unpaid": 0.0,
                 "risk_account_amount": 0.0,
                 "risk_account_amount_unpaid": 0.0,
             }
         )
         AccountMoveLine = self.env["account.move.line"].sudo()
-        customers = self.filtered(lambda x: x.customer and not x.parent_id)
+        customers = self.filtered(lambda x: x.customer_rank > 0 and not x.parent_id)
```

Comment:
I wouldn't restrict to those with that rank, as you can have invoices and not being classified as customer.

Reply:
Well, by default a partner has customer_rank = 0, right. But just when you have an invoice with this partner, then automatically customer_rank is updated and treat this partner as a customer.

Reply:
And what about sales order risk? You should remove this part of the domain and compute it for all partners no matter their rank.

Reply:
When you create a partner from customer the rank is 1:
https://github.com/odoo/odoo/blob/0274d61a11c6f5220d6af37a38adafb3e5276224/addons/account/models/partner.py#L500-L506
https://github.com/odoo/odoo/blob/80dd40dd86484b3ab8c47b3baa315843f0027871/addons/account/views/partner_view.xml#L233

I think that It is the least bad solution, Any suggestion?

Reply:
But what if you have created from other part (Contacts for example). For me, it should be:
```suggestion
        customers = self.filtered(lambda x: not x.parent_id)
```

Reply:
And what about suppliers?
Not to mention "weird" things like a partner with parent marked as a company...

Reply:
What is the problem with "suppliers"? We are computing the risks. For partners without sales activity, the risk will be 0.

Reply:
That field is very easy to fix for rare cases.
The rest of workarrounds seem to me a worse solution.

Reply:
Why? This is not going to be mass-computed for all partners AFAIK. Only entering on the partner form or checking risk for an operation, it will be computed. Is there's something I'm missing?

Reply:
_search_risk_exception
https://github.com/OCA/credit-control/pull/67#discussion_r440759340

Reply:
OK, for that one can be considered, but in this case is not needed

Reply:
What about partner with parent marked as a company?

Reply:
I don't get you, but is this related to `customer_rank`? If not, please keep discussions separated and let's handle this and then you open the other topic.